overlord/hookstate/hookmgr.go

```diff
@@ -256,24 +273,30 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	if hookExists {
-		output, err := runHook(context, tomb)
-		if err != nil {
-			if hooksup.TrackError {
-				trackHookError(context, output, err)
-			}
-			err = osutil.OutputErr(output, err)
-			if hooksup.IgnoreError {
-				task.State().Lock()
-				task.Errorf("ignoring failure in hook %q: %v", hooksup.Hook, err)
-				task.State().Unlock()
-			} else {
-				if handlerErr := context.Handler().Error(err); handlerErr != nil {
-					return handlerErr
-				}
-
-				return fmt.Errorf("run hook %q: %v", hooksup.Hook, err)
+	// some hooks get hijacked, e.g. the core configuration
+	var output []byte
+	if f := m.hijacked(hooksup.Hook, hooksup.Snap); f != nil {
+		context.Lock()
+		f(context)
```

Comment:
Shouldn't this have the opportunity to return an error, which is logged as usual into the handler?